### DISCUSSION TITLE
Problem between checks

### DISCUSSION THREAD

**Discussion by afuno** [sunny/actor#94] [Ideas]
Now there is a problem between validations. For example, between `type` and `inclusion`. If the type does not match what is required, then it will not be checked, since only `inclusion` will be checked. That is, only the last rule will be checked.

I think it is necessary to implement logic that would use all the described checks for `input` and `output`. This should share the responsibility and simplify the code, getting rid of unnecessary conditions inside `is`, for example, for `must`, inside which there is sometimes logic for checking for nil and type.

**Comment by sunny** [comment]
Hey @afuno! That sounds like a bug, can you open an issue with an example that would show this behavior and explain how you would want it to work? 🙏🏻

**Reply by afuno** [reply]
Created issue: https://github.com/sunny/actor/issues/95
